### PR TITLE
KOGITO-4639 Setup BDD tests coverage

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-4639

- build of a new profiling image (with `got test`) to be able to catch the coverage
- kustomization changed a bit to include the construction of profiling files
- need to have a specific data-access pod to get the coverage as the cover file will be created only after operator is stopped. so using a pvc to store the coverage and to retrieve it with data-access pod
- profiling is just an option of BDD tests (and not default)
- profiling is on PR. nightly will do the testing with the correct operator built.

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains a link to the JIRA issue
- [ ] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by radtriste** [comment]
thinking about having a full BDD tests nightly run with profiling (but not as part of the nightly process) to have a full coverage idea

**Comment by sutaakar** [comment]
@radtriste Wouldn't it have more sense to set ReadWriteMany for PVC and deploy the data access pod at the same time as deploying Kogito operator? That way we remove the risk of PVC being recycled in the time between deletion of Kogito operator and creating data access pod.
Otherwise great work.

**Review by sutaakar — APPROVED** [review]

**Comment by radtriste** [comment]
jenkins test

File: Makefile [hunk 1]
```diff
@@ -108,7 +122,7 @@ ifeq (, $(shell which kustomize))
 	KUSTOMIZE_GEN_TMP_DIR=$$(mktemp -d) ;\
 	cd $$KUSTOMIZE_GEN_TMP_DIR ;\
 	go mod init tmp ;\
-	go get sigs.k8s.io/kustomize/kustomize/v3@v3.5.4 ;\
+	go get sigs.k8s.io/kustomize/kustomize/v4@v4.0.5 ;\
```
Comment (ricardozanini) [inline Makefile:125]: Have you tested the `make before-pr` with this new kustomize version? I'm assuming it's working, but they are known to break things between releases. :(

**Review by ricardozanini — COMMENTED** [review]
It seems really cool! I'm just worried about the kustomize changes. The SDK created the `default` profile. We might run into problems when upgrading the SDK version.

File: Makefile (hunk 1 above)
Comment (radtriste) [reply Makefile:125]: yeah ... did this change because it is the one used in GHAction PR check ... will run the `before-pr` to be sure

**Comment by radtriste** [comment]
the `default` is still existing, I just abstracted a bit more to be able to reuse. but indeed, let's see when we update. anyway we could still put again in `default` if needed

**Comment by radtriste** [comment]
> @radtriste Wouldn't it have more sense to set ReadWriteMany for PVC and deploy the data access pod at the same time as deploying Kogito operator? That way we remove the risk of PVC being recycled in the time between deletion of Kogito operator and creating data access pod.
> Otherwise great work.

As discussed, AWS cluster storageclass does not support the `ReadWriteMany`:
```
Failed to provision volume with StorageClass "gp2": invalid AccessModes [ReadWriteMany]: only AccessModes [ReadWriteOnce] are supported
```
so will stay with `ReadWriteOnce` and have the `data-access` pod started after.
anyway, we only delete the Kogito operator deployment and not the pvc so we should still have the data in pvc remaining when we start the `data-access` pod, there should not be any recycling

**Comment by Kevin-Mok** [comment]
> /jenkins test

It seems that your PR is having [this error](https://github.com/kiegroup/kogito-operator/pull/841#issuecomment-821730401) as well even after the weekend, and I reran my Jenkins test twice with the same issue. Any idea what could be causing it?

**Comment by radtriste** [comment]
ok looks like the send to codecov from RHBA jenkins is having some struggle ... maybe due to Openstack network.
From AWS Jenkins (PR check), upload command of results has no problem
From RHBA Jenkins (Profiling nightly), upload command is finishing with `400 Bad request`.
I guess there is a network issue ... Created https://issues.redhat.com/browse/KOGITO-4992 to investigate this. Otherwise, the rest is good

**Review by sutaakar — APPROVED** [review]
Looks good to me.

**Review by vaibhavjainwiz — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**